darts/models/forecasting/tft_model.py

```diff
@@ -0,0 +1,915 @@
+
+        static_context_expanded = self.expand_static_context(
+            context=self.static_context_grn(static_embedding),
+            timesteps=timesteps
+        )
+
+        embeddings_varying_encoder = {name: input_vectors_past[name] for name in self.encoder_variables}
+        embeddings_varying_encoder, encoder_sparse_weights = self.encoder_vsn(
+            x=embeddings_varying_encoder,
+            context=static_context_expanded[:, :encoder_length],
+        )
+
+        embeddings_varying_decoder = {name: input_vectors_future[name] for name in self.decoder_variables}
+        embeddings_varying_decoder, decoder_sparse_weights = self.decoder_vsn(
+            x=embeddings_varying_decoder,
+            context=static_context_expanded[:, encoder_length:],
+        )
+
+        # LSTM
+        # calculate initial state
+        input_hidden = self.static_context_initial_hidden_lstm(static_embedding).expand(
+            self.lstm_layers, -1, -1
+        )
+        input_cell = self.static_context_initial_cell_lstm(static_embedding).expand(
+            self.lstm_layers, -1, -1
+        )
+
+        # run local lstm encoder
+        encoder_out, (hidden, cell) = self.lstm_encoder(
+            x=embeddings_varying_encoder,
+            hx=(input_hidden, input_cell),
+            lengths=encoder_lengths,
+            enforce_sorted=False
+        )
+
+        # run local lstm decoder
+        decoder_out, _ = self.lstm_decoder(
+            x=embeddings_varying_decoder,
+            hx=(hidden, cell),
+            lengths=decoder_lengths,
+            enforce_sorted=False,
+        )
+
+        # post lstm GateAddNorm
+        lstm_out_encoder = self.post_lstm_encoder_gan(x=encoder_out, skip=embeddings_varying_encoder)
+        lstm_out_decoder = self.post_lstm_encoder_gan(x=decoder_out, skip=embeddings_varying_decoder)
+
+        lstm_out = torch.cat([lstm_out_encoder, lstm_out_decoder], dim=1)
+
+        # static enrichment
+        static_context_enriched = self.static_context_enrichment(static_embedding)
+        attn_input = self.static_enrichment_grn(
+            x=lstm_out,
+            context=self.expand_static_context(context=static_context_enriched, timesteps=timesteps)
+        )
+
+        # multi-head attention
+        attn_out, attn_out_weights = self.multihead_attn(
+            q=attn_input[:, encoder_length:],  # query only for predictions
+            k=attn_input,
+            v=attn_input,
+            mask=self.get_attention_mask(
+                encoder_lengths=encoder_lengths,
+                decoder_length=timesteps - encoder_length,
+                device=past_target.device
+            ),
+        )
+
+        # skip connection over attention
+        attn_out = self.post_attn_gan(x=attn_out, skip=attn_input[:, encoder_length:])
+
+        # position-wise feed-forward
+        out = self.positionwise_feedforward_grn(x=attn_out, context=None)
```

Comment:
Do you know if this is in the paper?

Reply:
Yes, see arrow below

![Screenshot 2021-11-05 at 08 42 47](https://user-images.githubusercontent.com/84317528/140475713-d70b8608-5e57-4241-824c-4577905860be.png)